src/QMCHamiltonians/QMCHamiltonian.cpp

```diff
@@ -488,48 +488,59 @@ QMCHamiltonian::Return_t QMCHamiltonian::evaluate(ParticleSet& P)
   return LocalEnergy;
 }
 
-void QMCHamiltonian::flex_evaluate(const std::vector<QMCHamiltonian*>& H_list, const std::vector<ParticleSet*>& P_list, std::vector<Return_t>& LocalEnergies) const
+void QMCHamiltonian::flex_evaluate(const RefVector<QMCHamiltonian>& H_list,
+                                   const RefVector<ParticleSet>& P_list,
+                                   std::vector<Return_t>& LocalEnergies)
 {
   if (H_list.size() > 1)
   {
     for (int iw = 0; iw < H_list.size(); iw++)
-      H_list[iw]->LocalEnergy = 0.0;
+      H_list[iw].get().LocalEnergy = 0.0;
 
-    for (int i = 0; i < H.size(); ++i)
+    for (int i = 0; i < H_list[0].get().H.size(); ++i)
     {
-      ScopedTimer local_timer(myTimers[i]);
+      ScopedTimer local_timer(H_list[0].get().myTimers[i]);
       const auto HC_list(extract_HC_list(H_list, i));
-      H[i]->mw_evaluate(HC_list, P_list);
+
+      // This lambda accomplishes two things
+      // 1. It makes clear T& and not std::reference<T> is desired removing need for gets.
```

Comment:
Should this be std::ref<T> or std::reference_wrapper<T>?

Reply:
should be reference_wrapper an important disambiguation here.